test/jdk/javax/swing/JComboBox/JComboBoxScrollFocusTest.java

```diff
@@ -0,0 +1,155 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Component;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JScrollBar;
+import javax.swing.JScrollPane;
+import javax.swing.SwingUtilities;
+import javax.swing.plaf.basic.BasicComboPopup;
+
+/*
+ * @test
+ * @key headful
+ * @bug 6672644
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @requires os.family != "mac"
+ * @summary Tests JComboBox scrollbar behavior when alt-tabbing
+ * @run main JComboBoxScrollFocusTest
+ */
+
+public class JComboBoxScrollFocusTest {
+    private static Robot robot;
+    private static JFrame comboboxFrame;
+    private static JComboBox<String> combobox;
+
+    public static void main(String[] args) throws Exception {
+        robot = new Robot();
+        try {
+            SwingUtilities.invokeAndWait(JComboBoxScrollFocusTest::createAndShowGUI);
+            doTest();
+        } finally {
+            SwingUtilities.invokeAndWait(comboboxFrame::dispose);
+        }
+    }
+
+    private static void createAndShowGUI() {
+        comboboxFrame = new JFrame("JComboBoxScrollFocusTest Test Frame");
+        combobox = new JComboBox<>();
+        for (int i = 0; i < 100; i++) {
+            combobox.addItem(String.valueOf(i));
+        }
+        comboboxFrame.add(combobox);
+        comboboxFrame.setSize(400, 200);
+        comboboxFrame.setLocationRelativeTo(null);
+        comboboxFrame.setVisible(true);
+    }
+
+    static Rectangle getOnScreenBoundsOnEDT(Component component)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        robot.waitForIdle();
+        FutureTask<Rectangle> task = new FutureTask<>(()
+                -> new Rectangle(component.getLocationOnScreen(),
+                component.getSize()));
+        SwingUtilities.invokeLater(task);
+        return task.get(500, TimeUnit.MILLISECONDS);
+    }
+
+    private static int getScrollbarValue()
+            throws InterruptedException, InvocationTargetException,
+            ExecutionException, TimeoutException {
```

Comment:
Can be simplified to throw generic exception 
```suggestion
            throws Exception {
```